src/main.rs

```diff
@@ -0,0 +1,188 @@
+
+    // Source colorscheme script
+    // Wait for script to fully execute before continuing
+    let mut child = Command::new("/bin/bash").arg(base16_shell_colorscheme_path).spawn()?;
+    let status = child.wait()?;
+    if !status.success() {
+        eprintln!("Command finished with a non-zero status.");
+        return Err(
+            std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Script execution failed"
+            )
+        );
+    }
+
+    // Hooks
+    // Set env variables for hooks and then execute .sh hooks
+    // -----------------------------------------------------------------
+    let mut base16_shell_hooks_path = env::var("BASE16_SHELL_HOOKS_PATH").unwrap_or_default();
+    if base16_shell_hooks_path.is_empty() || !Path::new(&base16_shell_hooks_path).is_dir() {
+        base16_shell_hooks_path = format!("{}/hooks", base16_shell_path.display());
+
+        env::set_var("BASE16_SHELL_HOOKS_PATH", &base16_shell_hooks_path);
+    }
+
+    env::set_var("BASE16_SHELL_THEME_NAME_PATH", base16_shell_theme_name_path);
+    env::set_var("BASE16_CONFIG_PATH", base16_config_path);
+
+    let base16_shell_hooks_path = base16_shell_path.join("hooks");
+    if base16_shell_hooks_path.is_dir() {
+        for entry in fs::read_dir(base16_shell_hooks_path)? {
+            let entry = entry?;
+            let path = entry.path();
+
+            // Check if the file name ends with .sh
+            if path.extension().and_then(|ext| ext.to_str()) == Some("sh") {
+                Command::new("/bin/bash").arg(path).status()?;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    if args.len() < 2 {
+        eprintln!("You didn't provide a theme name argument.");
+        std::process::exit(1);
+    }
+
+    let config_path: PathBuf = env::var("XDG_CONFIG_HOME")
+        .map(PathBuf::from)
+        .unwrap_or_else(|_| {
+            let home = env::var("HOME").expect("HOME is not set");
+
+            PathBuf::from(home).join(".config")
+        });
+    let config_path_something = config_path.as_path();
+    let base16_config_path = config_path_something.join("tinted-theming");
+    let base16_shell_path: PathBuf = env::current_dir().expect("Failed to get current directory");
+    let base16_shell_colorscheme_path = base16_config_path.join("base16_shell_theme");
+    let base16_shell_theme_name_path = base16_config_path.join("theme_name");
+    let theme_name = &args[1];
+    let theme_script_path = base16_shell_path.join(format!("scripts/base16-{}.sh", theme_name));
+
+    if !theme_script_path.exists() {
+        eprintln!("Theme \"{}\" does not exist, try a different theme", theme_name);
+        std::process::exit(1);
+    }
+
+    if let Err(e) = ensure_config_files_exist(base16_config_path.as_path(), base16_shell_theme_name_path.as_path()) {
+        eprintln!("Error: {}", e);
+    }
+
+    match set_theme(
```

Comment:
It might be cleaner to make main() return a Result<(), anyhow::Error> and just propagate the errors up.